doc/lua_api.txt

```diff
@@ -2896,6 +2896,23 @@ For the following functions `x` can be either a vector or a number:
 * `vector.divide(v, x)`:
     * Returns a scaled vector or Schur quotient.
 
+For the following functions `a` is an angle in radians and `r` is a rotation
+vector ({x = <pitch>, y = <yaw>, z = <roll>}) where pitch, yaw and roll are
+angles in radians.
+
+* `vector.rotate(v, r)`:
+    * Applies the rotation `r` to `v` and returns the result.
+    * `vector.rotate({x = 0, y = 0, z = 1}, r)` and
+      `vector.rotate({x = 0, y = 1, z = 0}, r)` return vectors pointing
+      forward and up relative to an entity's rotation `r`.
+* `vector.rotate_around_axis(v1, v2, a)`:
+    * Returns `v1` rotated around axis `v2` by `a` radians according to
+      the left hand rule.
+* `vector.directions_to_rotation(direction, up)`:
+    * Returns a rotation vector for `direction` pointing forward using `up`
+      as the up vector.
+    * `direction` and `up` need to be unit vectors in a 90 degree angle to each other.
```

Comment:
This makes this function not very useful - you'd only be able to end up with a single axis of rotation. Unless the docs are wrong?

Reply:
I don't see an issue, `direction=(1, 0, 0) up=(0, 1, 0)`  is not the only way to make a 90° angle

Reply:
Let's say my direction is (1, 1, 0) and up is (0, 1, 0). They aren't at right angles, and this makes this function not very nice to use

Reply:
Maybe I misunderstand the point of this function. If up is (0, 1, 0), then this function should be equivalent to:

```lua
local function direction_to_rotation(direction)
	return {
		x = math.atan2(direction.y, math.sqrt(direction.z*direction.z+direction.x*direction.x)),
		y = -math.atan2(direction.x, direction.z),
		z = 0
	}
end
```

Reply:
I guess you'd need to rotate up along the plane formed by the original up and the direction, to reach a right angle

Reply:
What do you think about just returning zero as roll if 'up' is omitted?